task_templates/pipelines/python3_calibrated_anomaly_detection/custom.py

```diff
@@ -1,40 +1,110 @@
-    kwargs: Added for forwards compatibility
+        row_weights, during smart downsampling and when weights are explicitly specified in the project settings.
+    kwargs
+        Added for forwards compatibility.
+
     Returns
     -------
-    Nothing
+    None
+        fit() doesn't return anything, but must output an artifact
+        (typically containing a trained object) into output_dir
+        so that the trained object can be used during scoring.
     """
     estimator = make_anomaly()
 
     estimator.fit(X)
-    # You must serialize out your model to the output_dir given, however if you wish to change this
-    # code, you will probably have to add a load_model method to read the serialized model back in
-    # When prediction is done.
-    # Check out this doc for more information on serialization https://github.com/datarobot/custom-\
-    # model-templates/tree/master/custom_model_runner#python
-    # NOTE: We currently set a 10GB limit to the size of the serialized model
+
+    # Dump the trained object [in this example - a trained one class SVM model]
+    # into an artifact [in this example - artifact.pkl]
+    # and save it into output_dir so that it can be used later when scoring data
+    # Note: DRUM will automatically load the model when it is in the default format (see docs)
+    # and there is only one artifact file
     with open("{}/artifact.pkl".format(output_dir), "wb") as fp:
         pickle.dump(estimator, fp)
+
+
+def score(data: pd.DataFrame, model: Any, **kwargs: Dict[str, Any]) -> pd.DataFrame:
+    """
+    DataRobot will run this hook when the task is used for scoring inside a blueprint
+
+    This hook defines the output of a custom estimator and returns predictions on input data.
+    It should be skipped if a task is a transform.
+
+    Note: While best practice is to include the score hook, if the score hook is not present DataRobot will
+    add a score hook and call the default predict method for the library
+    See https://github.com/datarobot/datarobot-user-models#built-in-model-support for details
+
+    Parameters
+    ----------
+    data: pd.DataFrame
+        Is the dataframe to make predictions against. If the `transform` hook is utilized,
+        `data` will be the transformed data
+    model: Any
+        Trained object, extracted by DataRobot from the artifact created in fit().
+        In this example, contains trained sklearn pipeline extracted from artifact.pkl.
+    kwargs:
+        Additional keyword arguments to the method
+
+    Returns
+    -------
+    This method should return predictions as a dataframe with the following format:
+      Binary Classification: must have columns for each class label with floating- point class
+        probabilities as values. Each row should sum to 1.0. The original class names defined in the project
+        must be used as column names.
+      Regression: must have a single column called `Predictions` with numerical values
```

Comment:
We should update this to only reflect AD output

Reply:
done